Fix/school map view

Matinj031 opened the pull request.

Matinj031 committed:
feat(schoolDetailsModal): Adds school details modal component
Adds a new component to display school details in a modal.

This modal provides a user-friendly way to view comprehensive information about a school, including name, location, score, and last updated date. It includes separate layouts for desktop and mobile, optimizing the viewing experience across different devices. Also handles loading and error states.

Matinj031 committed:
feat(school/map): Improves school marker click handling

Updates the map marker click event to emit a custom event with school data.

This allows parent components to handle navigation or display school details, providing more flexibility and control over the user experience. It also ensures the map component remains focused on map interactions, delegating school-specific actions to other parts of the application.

Matinj031 committed:
feat(pages/school/index): Implements school details modal and map enhancements

Adds a modal to display school details when a marker is clicked on the map.
This allows users to quickly view information about a school without navigating
to a separate page.

Improves map view by initializing filter parameters and clearing school list
when switching to map view for performance.

Refactors school list fetching to handle both list and map views, using
location-based parameters for map view and filter-based parameters for list view.

Updates sorting logic to handle null or empty sort parameters gracefully.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Improves filter list display on mobile

Ensures the filter list is displayed correctly on mobile devices, even when the map is expanded.

Removes unnecessary CSS styles related to map view mode, simplifying the stylesheet.

Matinj031 committed:
Removes the top margin from the footer.

Matinj031 committed:
Removes map container style

Removes the map container style and related class.

The map container's styling is no longer needed in the component, as it's likely being handled elsewhere or is no longer relevant.

Matinj031 committed:
Adjusts school details modal position

Repositions the school details modal to appear higher on the screen.

The modal is now centered on both axes, improving its visibility and user experience.

Matinj031 committed:
Adjusts school list view layout

Refactors the school list view to occupy the full available space.

Removes specific height and width calculations, allowing for more responsive scaling.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Improves the display of school details by showing "New" instead of "N/A" when a school's score is not available.

Matinj031 committed:
Updates the school score display to show "New" when the score is unavailable.

Matinj031 committed:
Updates the school details navigation to open the link in a new tab, improving user experience by allowing users to keep the current page open.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Refactors leaflet plugin to TypeScript

Matinj031 committed:
Adds leaflet.markercluster typings

Matinj031 committed:
feat(school map component): Improves school map marker handling

Enhances the school map component by validating marker data, improving marker click handling, and adding error handling.

- Ensures all required school data is present before creating markers.
- Improves marker click handling to reliably access school data and emit a 'school-marker-clicked' event.
- Introduces error handling to catch invalid school data or other exceptions during marker click events, emitting a 'school-marker-click-error' event with relevant context for debugging.

Matinj031 committed:
Improves school details modal error handling

Enhances the school details modal to provide a better user experience when school data fails to load.

Matinj031 committed:
Improves school data handling on marker click

Enhances the school data handling when a marker is clicked on the map.
It adds validation to ensure data integrity and fetches additional details
if necessary, improving the user experience by ensuring complete
information is displayed in the modal. It also handles potential errors
during marker clicks, providing a fallback to direct navigation.

Adds a handler for school marker click errors and improves data validation for
displaying school details.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

Matinj031 committed:
Refactors the school details modal to improve navigation and user experience.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into fix/school-map-view

alireza013013 commented:
Hi @Matinj031 ,

Thank you for the changes you made to the map and the school list page, which improved their performance. However, there are three points I’d like to mention that I believe could make them even better:

1-Scroll Issue on the Page:
If you’ve noticed, the page currently scrolls, whereas it shouldn’t. This is also why the footer isn’t being displayed. After checking, I found the issue stems from the style applied to the div of the map component, which is currently:
`style="height: 100vh; width: 100vw"`
Changing it to the following should resolve the problem:
`style="height: 100%; width: 100%"`

2-List Component Refactoring:
There used to be a component called "List" that simply displayed the school list and emitted an event when reaching the end to fetch new data. However, there are now two separate components for desktop and mobile. I noticed that all the code is identical except for the school card section. If we can create a separate component just for the school card and merge the two list components back into one, I think the code would be much cleaner and more optimized. If you have any thoughts on this, I’d appreciate your input.

3-Styling Issue in Development Environment:
There’s a problem where the cards load with the new styles at first, but when the page mounts, they revert to the old styles.

Let me know what you think!

Matinj031 commented:
> Hi @Matinj031 ,
> 
> Thank you for the changes you made to the map and the school list page, which improved their performance. However, there are three points I’d like to mention that I believe could make them even better:
> 
> 1-Scroll Issue on the Page: If you’ve noticed, the page currently scrolls, whereas it shouldn’t. This is also why the footer isn’t being displayed. After checking, I found the issue stems from the style applied to the div of the map component, which is currently: `style="height: 100vh; width: 100vw"` Changing it to the following should resolve the problem: `style="height: 100%; width: 100%"`
> 
> 2-List Component Refactoring: There used to be a component called "List" that simply displayed the school list and emitted an event when reaching the end to fetch new data. However, there are now two separate components for desktop and mobile. I noticed that all the code is identical except for the school card section. If we can create a separate component just for the school card and merge the two list components back into one, I think the code would be much cleaner and more optimized. If you have any thoughts on this, I’d appreciate your input.
> 
> 3-Styling Issue in Development Environment: There’s a problem where the cards load with the new styles at first, but when the page mounts, they revert to the old styles.
> 
> Let me know what you think!

Hi @alireza013013 👋

Thank you so much for taking the time to review the changes and provide such detailed feedback! I really appreciate your thorough analysis and constructive suggestions.

Great news - I've already addressed points 1 and 3 that you mentioned on https://github.com/GamaEdtech/gamatrain-front/pull/459 :

✅ **Scroll Issue**: Fixed the map component styling by changing from `height: 100vh; width: 100vw` to `height: 100%; width: 100%` as you suggested. The footer should now display properly without the unwanted scrolling.

✅ **Styling Issue in Development**: Resolved the problem where cards were reverting to old styles after page mount.

Regarding **point 2 (List Component Refactoring)**: You make an excellent point about code optimization! Creating a separate school card component and merging the desktop/mobile list components back into one would definitely make the codebase cleaner and more maintainable. I think this is a great approach that follows the DRY principle.

I'd love to implement this refactoring in a follow-up task. Would you prefer to handle this optimization yourself, or would you like me to work on it? Either way works for me - just let me know your preference!

Thanks again for the valuable feedback and for helping make the code better. Your attention to detail is much appreciated! 🙏

Best regards

alireza013013 commented:
Hi @Matinj031 
Thank you for your complete explanation.
It's great that the problems have been resolved.
Considering that you implemented the card styles, I would appreciate it if you could also do this optimization part.
If I can be of any help, I would be happy to help.


Best regards
